client-cli/src/main/java/org/eclipse/dataspaceconnector/registration/cli/RegistrationServiceCli.java

```diff
@@ -63,7 +82,32 @@ private void init() {
         } catch (IOException e) {
             throw new RuntimeException("Error reading file " + privateKeyFile, e);
         }
-        var apiClient = ClientUtils.createApiClient(service, clientDid, privateKeyData);
-        this.registryApiClient = new RegistryApi(apiClient);
+
+        // temporary to preserve the backwards compatibility
+        if (dataspaceDid.isEmpty()) {
+            var apiClient = createApiClient(service, clientDid, privateKeyData);
+            this.registryApiClient = new RegistryApi(apiClient);
+            return;
+        }
+
+        registryApiClient = new RegistryApi(createApiClient(registrationUrl(), clientDid, privateKeyData));
+    }
+
+    private String registrationUrl() {
+        var didWebResolver = new WebDidResolver(httpClient(), useHttpsScheme(), MAPPER, new ConsoleMonitor());
```

Comment:
instantiating a `new ConsoleMonitor()` should not be the in the purview of a method that resolves a URL. Move to field or similar?

Reply:
moved to field